libraries/opensk/src/api/persist.rs

```diff
@@ -386,6 +418,102 @@ pub trait Persist {
         }
     }
 
+    /// Lists all stored template IDs.
+    #[cfg(feature = "fingerprint")]
+    fn template_infos(&self) -> CtapResult<Vec<TemplateInfo>> {
+        let mut id_list = Vec::new();
+        for key in keys::FRIENDLY_NAMES {
+            if let Some(data) = self.find(key)? {
+                let cbor_value = cbor_read(&data)
+                    .map_err(|_| Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR)?;
+                let template_info = TemplateInfo::try_from(cbor_value)?;
+                id_list.push(template_info);
+            }
+        }
+        Ok(id_list)
+    }
+
+    /// Stores a new template ID.
+    ///
+    /// Returns `Err(CTAP2_ERR_FP_DATABASE_FULL)` if the storage is full.
+    #[cfg(feature = "fingerprint")]
+    fn store_template_id(&mut self, template_id: &[u8]) -> CtapResult<()> {
+        for key in keys::FRIENDLY_NAMES {
+            if self.find(key)?.is_some() {
+                continue;
+            }
+            let template_info = TemplateInfo {
+                template_id: template_id.to_vec(),
+                template_friendly_name: None,
+            };
+            let mut data = Vec::new();
+            cbor_write(cbor::Value::from(template_info), &mut data)?;
+            self.insert(key, &mut data)?;
```

Comment:
The `mut` seems weird.
```suggestion
            self.insert(key, &data)?;
```

Reply:
I was surprised that this wasn't caught by clippy, and indeed, the new feature wasn't covered yet. Fixed all other clippy findings too.